implementations/code_df/commit.py

```diff
@@ -17,73 +19,87 @@ class Commit(Metric):
         would mean that all commits from the first commit to the commit
         who last falls inside the until range will be included.
 
-    :param issrccode_obj: A reference to IsSourceCode class.
+    :param is_code:  list of CodeCondition objects
         It is used to determine what comprises source code.
         """
 
-    def __init__(self, items, date_range=(None, None), issrccode_obj=None):
+    def __init__(self, items, date_range=(None, None), is_code=[conditions.Naive()],
+                 conds=[]):
```

Comment:
@jgbarah, it would be better to set `conds` to `None`. [The problem.](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument)

Reply:
Hmmmm. Thanks for the suggestion. However, I think that only applies if you change `conds` within the function, which is something yo don't do in this case, and would make little sense doing. Or maybe I'm wrong? Can you suggest some example where this would be a problem?

Reply:
@jgbarah, I agree. `conds`, as well as `is_code` parameters are not getting modified. It is OK to leave it there. Or you could do something like this inside `__init__`:
```python
if cond is None:
    cond = list()
```